bin/node/cli/src/command.rs

```diff
@@ -157,7 +157,14 @@ pub fn run() -> Result<()> {
 			let runner = cli.create_runner(cmd)?;
 			runner.async_run(|config| {
 				let PartialComponents { client, task_manager, backend, .. } = new_partial(&config)?;
-				Ok((cmd.run(client, backend), task_manager))
+				let client_clone = client.clone();
+				let revert_aux = Box::new(move |blocks| {
+					sc_consensus_babe::revert(client_clone, blocks)?;
+					// TODO: grandpa
+					Ok(())
+				});
```

Comment:
A pattern I have used with success for such clone-before-capture variables was:

```rust
let revert_aux = Box::new({
  let client = client.clone();
  move |blocks| {
    sc_consensus_babe::revert(client, blocks)?;
    // TODO: grandpa
    Ok(())
  }
});
```

I do not say it is better in all aspects, just wanted to share it as an idea.

Reply:
Good to know. But currently the clone has been moved in the closure caller.